src/HeadlessUniversalCheckout/Managers/VaultManager.ts

```diff
@@ -0,0 +1,98 @@
+        return new Promise(async (resolve, reject) => {
+            try {
+                await RNPrimerHeadlessUniversalCheckoutVaultManager.configure();
+                resolve();
+            } catch (err) {
+                console.error(err);
+                reject(err);
+            }
+        });
+    }
+
+    async fetchVaultedPaymentMethods(): Promise<VaultedPaymentMethod[]> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                const data: PrimerVaultedPaymentMethodResult =
+                    await RNPrimerHeadlessUniversalCheckoutVaultManager.fetchVaultedPaymentMethods();
+                const paymentMethods: VaultedPaymentMethod[] = data.paymentMethods;
+                resolve(paymentMethods);
+            } catch (err) {
+                console.error(err);
+                reject(err);
+            }
+        });
+    }
+
+    async deleteVaultedPaymentMethod(vaultedPaymentMethodId: String): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                await RNPrimerHeadlessUniversalCheckoutVaultManager.deleteVaultedPaymentMethod(vaultedPaymentMethodId);
+                resolve();
+            } catch (err) {
+                console.error(err);
+                reject(err);
+            }
+        });
+    }
+
+    async validate(vaultedPaymentMethodId: String, additionalData: VaultedPaymentMethodAdditionalData): Promise<ValidationError[]> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                const data: PrimerValidationErrorResult =
+                    await RNPrimerHeadlessUniversalCheckoutVaultManager.validate(vaultedPaymentMethodId, JSON.stringify(additionalData));
+                const errors: ValidationError[] = data.validationErrors;
+                resolve(errors);
+            } catch (err) {
+                console.error(err);
+                reject(err);
+            }
+        });
+    }
+
+    async startPaymentFlow(vaultedPaymentMethodId: String): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                await RNPrimerHeadlessUniversalCheckoutVaultManager.startPaymentFlow(vaultedPaymentMethodId);
+                resolve();
+            } catch (err) {
+                console.error(err);
+                reject(err);
+            }
+        });
+    }
+
+    async startPaymentFlowWithAdditionalData(vaultedPaymentMethodId: String, additionalData: VaultedPaymentMethodAdditionalData): Promise<void> {
+        return new Promise(async (resolve, reject) => {
+            try {
+                await RNPrimerHeadlessUniversalCheckoutVaultManager.startPaymentFlowWithAdditionalData(vaultedPaymentMethodId, JSON.stringify(additionalData));
+                resolve();
+            } catch (err) {
+                console.error(err);
+                reject(err);
+            }
+        });
+    }
+}
```

Comment:
@HarryPrimer Aladin is stating that these 2 functions should become one functuon due to the nature of JS syntax.

Something like this. 
<img width="580" alt="Screenshot 2023-12-08 at 11 31 08" src="https://github.com/primer-io/primer-sdk-react-native/assets/137198782/3465748b-1003-42d2-9d80-61836a6f7a6f">

Reply:
Okay I'll make this change and reflect it in the docs